zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -32,8 +46,9 @@ pub fn init_wallet_db<P>(wdb: &WalletDb<P>) -> Result<(), rusqlite::Error> {
     wdb.conn.execute(
         "CREATE TABLE IF NOT EXISTS accounts (
             account INTEGER PRIMARY KEY,
-            extfvk TEXT NOT NULL,
-            address TEXT NOT NULL
+            extfvk TEXT,
+            address TEXT,
+            transparent_address TEXT
```

Comment:
In a pairing, @nuttycom told me that the migration code was implemented directly in the mobile SDKs. This is insufficient for us, because `zcash_client_sqlite` can (in theory) have users beyond the mobile SDKs.

The resolution we came to is that we will require that a schema migration facility be built (with some kind of detection in `init_wallet_db` that returns an error if the database already exists but needs a migration) before the next release of `zcash_client_sqlite`. This unblocks this PR (and unblocks `zcashd`'s usage of it), but means we will not be cutting a release of `zcash_client_{backend/sqlite}` when we cut the `zcash_primitives` etc crates (which themselves will be after 4.7.0).

Reply:
This is now https://github.com/zcash/librustzcash/issues/489

Reply:
Yes we created db migration code to add the `transparent_address` field. The had do the manual SQLite "add column" this means programmatically doing rename table -> create new table with additional column(s) -> copy the data over.

Then we would for each account derive a transparent_address.

As for clients using this, we know that Nighthawk and Unstoppable have clients deployed. There are probably more around from the Zcash forks that are out there that probably have forks of everything.

I agree with the SDKs not being the ones migrating the data. since they are not the ones initializing it. On Android, there might be an ownership conflict due to the usage or the Pseudo-ORM library called Room. On iOS we are using a vanilla swift SQLite wrapper that doesn't do anything else than providing a decent API and some thread-safety serial queue access to the file handle.